runtime/bindings/python/iree/runtime/io.py

```diff
@@ -64,3 +67,137 @@ def save_archive_file(entries: dict[str, Union[Any, SplatValue]], file_path: Pat
         else:
             index.add_buffer(key, value)
     index.create_archive_file(str(file_path))
+
+
+def parameter_index_add_numpy_ndarray(
+    index: ParameterIndex, name: str, array: np.ndarray
+):
```

Comment:
This file already has `import array`, so this function shadows `array` with a new local variable. 

We have similar code that doesn't also `import array`: https://github.com/iree-org/iree/blob/a31da1f7ab2c81d3fd6eb74f3950d73b56607852/runtime/bindings/python/iree/runtime/system_api.py#L80-L88

I guess this is fine, just be careful :)